unstable test TestAzblob

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
 --- FAIL: TestAzblob (8.13s)
    azblob_test.go:68: 
        	Error Trace:	br/pkg/storage/azblob_test.go:68
        	Error:      	Received unexpected error:
        	            	Put "http://127.0.0.1:10000/devstoreaccount1/test?restype=container": net/http: HTTP/1.x transport connection broken: malformed HTTP status code "status"
        	            	Failed to create container
        	            	github.com/pingcap/tidb/br/pkg/storage.createContainer
        	            		br/pkg/storage/azblob_test.go:51
        	            	github.com/pingcap/tidb/br/pkg/storage.TestAzblob
        	            		br/pkg/storage/azblob_test.go:63
        	            	testing.tRunner
        	            		GOROOT/src/testing/testing.go:1576
        	            	runtime.goexit
        	            		src/runtime/asm_amd64.s:1598
        	Test:       	TestAzblob 
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/br/pkg/storage/azblob_test.go
+++ b/br/pkg/storage/azblob_test.go
@@ -61,7 +61,7 @@ func TestAzblob(t *testing.T) {
 	}
 	builder := &sharedKeyAzuriteClientBuilder{}
 	skip, err := createContainer(ctx, builder, options.Bucket)
-	if skip {
+	if skip || err != nil {
 		t.Log("azurite is not running, skip test")
 		return
 	}

